x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1277,13 +1278,21 @@ public Collection<?> createComponents(PluginServices services) {
             new MlAutoscalingDeciderService(memoryTracker, settings, nodeAvailabilityZoneMapper, clusterService)
         );
 
-        MlInitializationService mlInitializationService = new MlInitializationService(
-            settings,
+        AdaptiveAllocationsScalerService adaptiveAllocationsScalerService = new AdaptiveAllocationsScalerService(
```

Comment:
Note: `AdaptiveAllocationsScalerService` is turned into a component, so that it can be injected in `TransportExternalInferModelAction`.